{
    "containers": {
        "cna": {
            "affected": [
                {
                    "product": "httplib2",
                    "vendor": "httplib2",
                    "versions": [
                        {
                            "status": "affected",
                            "version": "< 0.19.0"
                        }
                    ]
                }
            ],
            "descriptions": [
                {
                    "lang": "en",
                    "value": "httplib2 is a comprehensive HTTP client library for Python. In httplib2 before version 0.19.0, a malicious server which responds with long series of \"\\xa0\" characters in the \"www-authenticate\" header may cause Denial of Service (CPU burn while parsing header) of the httplib2 client accessing said server. This is fixed in version 0.19.0 which contains a new implementation of auth headers parsing using the pyparsing library."
                }
            ],
            "metrics": [
                {
                    "cvssV3_1": {
                        "attackComplexity": "LOW",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "HIGH",
                        "baseScore": 7.5,
                        "baseSeverity": "HIGH",
                        "confidentialityImpact": "NONE",
                        "integrityImpact": "NONE",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                        "version": "3.1"
                    }
                }
            ],
            "problemTypes": [
                {
                    "descriptions": [
                        {
                            "cweId": "CWE-400",
                            "description": "CWE-400 Uncontrolled Resource Consumption",
                            "lang": "en",
                            "type": "CWE"
                        }
                    ]
                }
            ],
            "providerMetadata": {
                "dateUpdated": "2021-02-08T19:45:19",
                "orgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
                "shortName": "GitHub_M"
            },
            "references": [
                {
                    "tags": [
                        "x_refsource_CONFIRM"
                    ],
                    "url": "https://github.com/httplib2/httplib2/security/advisories/GHSA-93xj-8mrv-444m"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/httplib2/httplib2/pull/182"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://github.com/httplib2/httplib2/commit/bd9ee252c8f099608019709e22c0d705e98d26bc"
                },
                {
                    "tags": [
                        "x_refsource_MISC"
                    ],
                    "url": "https://pypi.org/project/httplib2"
                }
            ],
            "source": {
                "advisory": "GHSA-93xj-8mrv-444m",
                "discovery": "UNKNOWN"
            },
            "title": "Regular Expression Denial of Service in httplib2",
            "x_legacyV4Record": {
                "CVE_data_meta": {
                    "ASSIGNER": "security-advisories@github.com",
                    "ID": "CVE-2021-21240",
                    "STATE": "PUBLIC",
                    "TITLE": "Regular Expression Denial of Service in httplib2"
                },
                "affects": {
                    "vendor": {
                        "vendor_data": [
                            {
                                "product": {
                                    "product_data": [
                                        {
                                            "product_name": "httplib2",
                                            "version": {
                                                "version_data": [
                                                    {
                                                        "version_value": "< 0.19.0"
                                                    }
                                                ]
                                            }
                                        }
                                    ]
                                },
                                "vendor_name": "httplib2"
                            }
                        ]
                    }
                },
                "data_format": "MITRE",
                "data_type": "CVE",
                "data_version": "4.0",
                "description": {
                    "description_data": [
                        {
                            "lang": "eng",
                            "value": "httplib2 is a comprehensive HTTP client library for Python. In httplib2 before version 0.19.0, a malicious server which responds with long series of \"\\xa0\" characters in the \"www-authenticate\" header may cause Denial of Service (CPU burn while parsing header) of the httplib2 client accessing said server. This is fixed in version 0.19.0 which contains a new implementation of auth headers parsing using the pyparsing library."
                        }
                    ]
                },
                "impact": {
                    "cvss": {
                        "attackComplexity": "LOW",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "HIGH",
                        "baseScore": 7.5,
                        "baseSeverity": "HIGH",
                        "confidentialityImpact": "NONE",
                        "integrityImpact": "NONE",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:N/I:N/A:H",
                        "version": "3.1"
                    }
                },
                "problemtype": {
                    "problemtype_data": [
                        {
                            "description": [
                                {
                                    "lang": "eng",
                                    "value": "CWE-400 Uncontrolled Resource Consumption"
                                }
                            ]
                        }
                    ]
                },
                "references": {
                    "reference_data": [
                        {
                            "name": "https://github.com/httplib2/httplib2/security/advisories/GHSA-93xj-8mrv-444m",
                            "refsource": "CONFIRM",
                            "url": "https://github.com/httplib2/httplib2/security/advisories/GHSA-93xj-8mrv-444m"
                        },
                        {
                            "name": "https://github.com/httplib2/httplib2/pull/182",
                            "refsource": "MISC",
                            "url": "https://github.com/httplib2/httplib2/pull/182"
                        },
                        {
                            "name": "https://github.com/httplib2/httplib2/commit/bd9ee252c8f099608019709e22c0d705e98d26bc",
                            "refsource": "MISC",
                            "url": "https://github.com/httplib2/httplib2/commit/bd9ee252c8f099608019709e22c0d705e98d26bc"
                        },
                        {
                            "name": "https://pypi.org/project/httplib2",
                            "refsource": "MISC",
                            "url": "https://pypi.org/project/httplib2"
                        }
                    ]
                },
                "source": {
                    "advisory": "GHSA-93xj-8mrv-444m",
                    "discovery": "UNKNOWN"
                }
            }
        }
    },
    "cveMetadata": {
        "assignerOrgId": "a0819718-46f1-4df5-94e2-005712e83aaa",
        "assignerShortName": "GitHub_M",
        "cveId": "CVE-2021-21240",
        "datePublished": "2021-02-08T19:45:19",
        "dateReserved": "2020-12-22T00:00:00",
        "dateUpdated": "2021-02-08T19:45:19",
        "state": "PUBLISHED"
    },
    "dataType": "CVE_RECORD",
    "dataVersion": "5.0"
}